# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/layers/ipc/TextureForwarder.h
# Commit: 4357d695b8d5
# Full Hash: 4357d695b8d5dd901788d922b73e89f16c45b2c9
# Author: Bob Owen <bobowencode@gmail.com>
# Date: 2019-06-07 16:20:31
# Regressor Bug: 1464032
# File Overlap Count: 1
# Description:
#   Bug 1464032 Part 12: Add CanvasParent, CanvasChild and RecordedTextureData. r=mattwoodrow, jld
#   
#   RecordedTextureData records TextureData calls for play back in the GPU process.
#   CanvasChild and CanvasParent set up the recorder and translator.
#   They also help to manage the starting of translation and co-ordinating the
# ==============================================================================

diff -r e32d6c3ba887 -r 4357d695b8d5 gfx/layers/ipc/TextureForwarder.h
--- a/gfx/layers/ipc/TextureForwarder.h	Sun Dec 02 14:17:24 2018 +0000
+++ b/gfx/layers/ipc/TextureForwarder.h	Sun Dec 02 14:19:11 2018 +0000
@@ -19,6 +19,7 @@
 class IShmemAllocator;
 }
 namespace layers {
+class CanvasChild;
 
 /**
  * An abstract interface for classes that implement the autogenerated
@@ -78,6 +79,11 @@
       LayersBackend aLayersBackend, TextureFlags aFlags, uint64_t aSerial,
       wr::MaybeExternalImageId& aExternalImageId,
       nsIEventTarget* aTarget = nullptr) = 0;
+
+  /**
+   * Returns the CanvasChild for this TextureForwarder.
+   */
+  virtual already_AddRefed<CanvasChild> GetCanvasChild() { return nullptr; };
 };
 
 }  // namespace layers